Updates needed in the workflow to accommodate recent EGA and cluster changes

pmb59 opened the pull request:
After some changes  in EGA and pyega3, and in our infrastructure, this workflow requires substantial changes to adapt and simplify the process:

- [x] Migrate executor from LSF to SLURM
- [x] There is no need to decrypt data files downloaded through the pyEGA3 download client. Download comes on fastq format
- [x] Pin latest version of `pyega3`
- [x] It seems to be possible to download data in Fastq directly, thus avoiding the bam-to-fastq conversion
- [x] Metadata downloads have also changed from XML+mapping files [to files in tsv/csv/json format ](https://ega-archive.org/access/download/metadata/#DatasetMappings). Changes in bin/arrange_data.R for the goal of retrieving the final `EGADXXXXXX.merged.tsv`
- [x] Migrate from DSL1 to DSL2
- [x] Deprecate Aspera fetching mode
- [x] Enable Seqera platform (for logging and monitoring)
- [x] Update README

pmb59 committed:
Update README.md

pmb59 committed:
migrate nextflow.config to SLURM

pmb59 committed:
pin versions of python and pyega3

pmb59 committed:
fix typo & rm tab

pmb59 committed:
update arrange_data.R

pmb59 committed:
update and simplify workflow

pmb59 committed:
fix indentation in main.nf

pmb59 committed:
add file with with some env variables

pmb59 committed:
update README

pmb59 committed:
README final edits

pmb59 committed:
delete comment & minor edit to release version

pmb59 committed:
Update README.md

Co-authored-by: Anil Thanki <thanki.anil@gmail.com>

jcmarca commented on main.nf:
```diff
@@ -1,337 +1,52 @@
-
+        mkdir -p ega_data
+        # list unencrypted md5 checksums for all files
+        pyega3 -cf $egaCredentialsDir/ega.credentials files ${params.EGA_DATASET_ID}
```
${params.EGA_DATASET_ID} references a parameter from the "params" object not the input value of the process EGA_DATASET_ID. 
Should that be changed by $EGA_DATASET_ID?

jcmarca commented on main.nf:
```diff
@@ -1,337 +1,52 @@
+        # list unencrypted md5 checksums for all files
+        pyega3 -cf $egaCredentialsDir/ega.credentials files ${params.EGA_DATASET_ID}
+        # download the dataset
+        pyega3 -c 10 -cf $egaCredentialsDir/ega.credentials fetch ${params.EGA_DATASET_ID} --output-dir ega_data --max-retries -1 --retry-wait 10
```
Same as previous comment

jcmarca commented on main.nf:
```diff
@@ -1,337 +1,52 @@
-        pyega3 -d -cf $egaCredentialsDir/ega.credentials fetch $fileId
-        mv $fileId/$fileName .
+        mkdir -p ega_metadata
+        ${workflow.launchDir}/bin/arrange_data.R -m $metadataDir -i ${params.EGA_DATASET_ID} -o ega_metadata/${params.EGA_DATASET_ID}.merged.csv
```
Same as previous comments

pmb59 committed:
edit readme

jcmarca reviewed (commented):
I have only reviewed main.nf and nextflow.config files.
I have write comments in some lines regarding the variable referenced using $params.
As an additional comment in the cextflow.config file I would check (if it has not been previously) that the parameters specified for the partition "production" align with the configuration of that partition in slurm

pmb59 commented on bin/arrange_data.R:
```diff
@@ -13,148 +18,232 @@ option_list = list(
-        }
-        multiple <- names(lengths)[lengths > 1]
-        if (length(multiple) > 0){
-            write(paste("The following match multiple files in downloads:", paste(multiple, collapse=', ')), stderr())
-        }
+metadata$library_strategy <- study_experiment_run_sample$library_strategy[matched_indices]
+metadata$library_source <- study_experiment_run_sample$library_source[matched_indices]
+metadata$library_layout <- study_experiment_run_sample$library_layout[matched_indices]
 
-        q(status=1)
-    }
+metadata$library_name <- study_experiment_run_sample$library_name[matched_indices]
+metadata$library_selection <- study_experiment_run_sample$library_selection[matched_indices]
+metadata$study_title <- study_experiment_run_sample$study_title[matched_indices]
+metadata$study_type <- study_experiment_run_sample$study_type[matched_indices]
+metadata$instrument_platform <- study_experiment_run_sample$instrument_platform[matched_indices]
+metadata$instrument_model <- study_experiment_run_sample$instrument_model[matched_indices]
```
It can't be used for all the chunk, as source and dest columns names are not equal, e.g. 
```
metadata$ega_run_id <- study_experiment_run_sample$run_accession_id[matched_indices]
metadata$ega_experiment_id <- study_experiment_run_sample$experiment_accession_id[matched_indices]
```
but I'll apply for the rest

pmb59 replied:
[45b4c98](https://github.com/ebi-gene-expression-group/ega_downloader/pull/9/commits/45b4c987f7b1d1bef8dd7d3b2c90459768c7cdf0)

anilthanki replied:
> It can't be used for all the chunk, as source and dest columns names are not equal, e.g.
> 
> ```
> metadata$ega_run_id <- study_experiment_run_sample$run_accession_id[matched_indices]
> metadata$ega_experiment_id <- study_experiment_run_sample$experiment_accession_id[matched_indices]
> ```
> 
> but I'll apply for the rest

Thats where it uses `columns` to go through all variables.. But happy with your solution as it minimises manual changes in future

pmb59 committed:
update arrange_data.R [refactor code]

pmb59 committed:
reference input value not params object in processes

pmb59 commented:
Thanks @anilthanki and @jcmarca for reviewing! 

Jose, I have addressed your comments here [2e784e4](https://github.com/ebi-gene-expression-group/ega_downloader/pull/9/commits/2e784e4267cba4484b807e790e5285a8f846e800).

anilthanki reviewed (approved):
LGTM
